Linux_scripts/rhui-check/rhui-check.py

```diff
@@ -0,0 +1,592 @@
+
+    black =    "\x1b[30;1m"
+    grey =     "\x1b[30;0m"
+    red =      "\x1b[31;20m"
+    green =    "\x1b[32;20m"
+    yellow =   "\x1b[33;20m"
+    bright_red = "\x1b[91;1m"
+    bold_red = "\x1b[31;1m"
+    bold_green =    "\x1b[32;1m"
+    bold_yellow =   "\x1b[33;1m"
+    reset =    "\x1b[0m"
+
+
+    # format = "%(asctime)s - %(name)s - %(levelname)s - %(message)s (%(filename)s:%(lineno)d)"
+    format = "%(asctime)s - %(name)s - %(levelname)s - %(message)s"
+
+    FORMATS = {
+        logging.DEBUG: black + format + reset,
+        logging.INFO: bold_green + format + reset,
+        logging.WARNING: bold_yellow + format + reset,
+        logging.ERROR: bright_red + format + reset,
+        logging.CRITICAL: bold_red + format + reset
+    }
+
+    def format(self, record):
+        log_fmt = self.FORMATS.get(record.levelno)
+        formatter = logging.Formatter(log_fmt)
+        return formatter.format(record)
+
+
+
+if os.geteuid() != 0:
+   logger.critical('This script needs to execute with root privileges')
+   logger.critical('You could leverage the sudo tool to gain administrative privileges')
+   exit(1)
+
+def start_logging(debug_level = False):
+    """This function sets up the logging configuration for the script and writes the log to /var/log/rhuicheck.log"""
+
+    logger = logging.getLogger(__name__)
+
+    console_handler = logging.StreamHandler()
+    color_formatter = CustomFormatter()
+    console_handler.setFormatter(color_formatter)    
+    logger.addHandler(console_handler)
+    console_handler.setLevel(logging.INFO)
+
+    if debug_level:
+        console_handler.setLevel(logging.DEBUG)
+
+    try:
+        log_filename = '/var/log/rhuicheck.log'
+        file_handler = logging.FileHandler(filename=log_filename)
+        plain_formatter = logging.Formatter("%(asctime)s - %(name)s - %(levelname)s - %(message)s")
+        file_handler.setFormatter(plain_formatter)    
+    except:
+        logger.critical("Unable to create log file in /var/log/rhuicheck.log, make sure the script is running with root privileges, the filesystem has enough space and it's not in Read-Only mode")
+        exit(1)
+    else:
+        file_handler.setLevel(logging.DEBUG)
+        logger.addHandler(file_handler)
+
+    logger.setLevel(logging.DEBUG)
+    return logger
+       
+parser = argparse.ArgumentParser()
+parser.add_argument(  '--debug','-d',
+                      action='store_true',
+                      help='Use DEBUG level')
+args = parser.parse_args()
+logger = start_logging(args.debug)
+
+try:
+    import requests
+except ImportError:
+    logger.critical("'requests' python module not found, but it's required for this test script, review your python installation.")
+    exit(1) 
+
+rhui3 = ['13.91.47.76', '40.85.190.91', '52.187.75.218']
+rhui4 = ['52.136.197.163', '20.225.226.182', '52.142.4.99', '20.248.180.252', '20.24.186.80']
+rhuius = ['13.72.186.193', '13.72.14.155', '52.224.249.194']
```

Comment:
There is no rhui in usgov anymore. We only have rhui4

Reply:
As clarified over chat, then old IPs are used to detect old hard-coded configurations and provide feedback. That validation is not going to be removed from the script as it's important to detect those old hard-coded configs that might cause multiple issues.